// Copyright 2020 Lisandro Bravo.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";


message movingActionMap {
  required int32 start_action = 1;
  enum actType { 
    TRASLATION = 1;
    ROTATION = 2;
  } 
  required actType action_type = 2;
  required int32 landmark_id = 3;
  required int32 angle_number = 4;
  required float action_threshold = 5; 
  required float time_between_actions = 6;
  required bool auto_repeat = 7 [default = true];
  optional int32 max_repeat = 8 [default = 5];
  required string topic = 9;
  required string positive_payload = 10;
  required string negative_payload = 11;
}


// Options to NonMaxSuppression calculator, which performs non-maximum
// suppression on a set of detections.
message movingDynamicGesturesCalculatorOptions {
  extend CalculatorOptions {
    optional movingDynamicGesturesCalculatorOptions ext = 55483323;
  }
  
  
  // Maximum time allowed between startind and endind gesture .
  optional double moving_time_out_s = 1 [default = 2.5];
  repeated movingActionMap moving_actions_map = 2;
  
}
